admin/class-primary-term-admin.php

```diff
@@ -263,7 +263,24 @@ private function map_taxonomies_for_js( $taxonomy ) {
 			$primary_term = '';
 		}
 
-		$terms = get_terms( $taxonomy->name );
+		$terms = get_terms(
+			[
+				'taxonomy'               => $taxonomy->name,
+				'update_term_meta_cache' => false,
+				'fields'                 => 'id=>name',
+			]
+		);
+
+		$mapped_terms_for_js = [];
```

Comment:
We don't often use closures nor do we often pass variables by reference within Yoast. I'm afraid many of our developers will not follow exactly what's going on here. Could you refactor this to use a foreach loop?